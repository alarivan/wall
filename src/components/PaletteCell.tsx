import styled from 'styled-components';
import Cell from './Cell';

interface PaletteCellProps {
  active?: boolean;
}

const PaletteCell = styled(Cell)<PaletteCellProps>`
  box-shadow: 1px 1px 5px 0px rgba(0, 0, 0, 0.75);
  border-style: solid;
  border-color: black;
  border-width: 0;
  border-radius: ${({ active }) => (active ? '50%' : 'none')};
  transition: border-radius 50ms ease-in-out;
  height: 100%;

  :hover {
    border-width: 0;
    border-radius: ${({ active }) => (active ? '50%' : 'none')};
  }
`;

export default PaletteCell;
