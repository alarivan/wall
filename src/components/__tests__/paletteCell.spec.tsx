import React from 'react';
import { render, cleanup } from '@testing-library/react';
import PaletteCell from '../PaletteCell';

afterEach(cleanup);

describe('PaletteCell', () => {
  it('renders correctly', () => {
    const { asFragment, container } = render(<PaletteCell cellColor='red' />);

    expect(container.firstChild).toHaveStyleRule('border-radius', 'none');
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with active', () => {
    const { asFragment, container } = render(
      <PaletteCell cellColor='red' active={true} />,
    );

    expect(container.firstChild).toHaveStyleRule('border-radius', '50%');

    expect(asFragment()).toMatchSnapshot();
  });
});
