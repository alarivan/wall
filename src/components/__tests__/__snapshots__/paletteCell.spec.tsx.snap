// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaletteCell renders correctly 1`] = `
<DocumentFragment>
  .c0 {
  background: red;
  -webkit-transition: -webkit-transform 100ms ease-in-out;
  -webkit-transition: transform 100ms ease-in-out;
  transition: transform 100ms ease-in-out;
  cursor: pointer;
  box-shadow: 1px 1px 5px 0px rgba(0,0,0,0.75);
  border-style: solid;
  border-color: black;
  border-width: 0;
  border-radius: none;
  -webkit-transition: border-radius 50ms ease-in-out;
  transition: border-radius 50ms ease-in-out;
}

.c0:hover {
  border-width: 1px;
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
}

.c0:hover {
  border-width: 0;
  border-radius: none;
}

<div
    class="c0"
  />
</DocumentFragment>
`;

exports[`PaletteCell renders correctly with active 1`] = `
<DocumentFragment>
  .c0 {
  background: red;
  -webkit-transition: -webkit-transform 100ms ease-in-out;
  -webkit-transition: transform 100ms ease-in-out;
  transition: transform 100ms ease-in-out;
  cursor: pointer;
  box-shadow: 1px 1px 5px 0px rgba(0,0,0,0.75);
  border-style: solid;
  border-color: black;
  border-width: 0;
  border-radius: 50%;
  -webkit-transition: border-radius 50ms ease-in-out;
  transition: border-radius 50ms ease-in-out;
}

.c0:hover {
  border-width: 1px;
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
}

.c0:hover {
  border-width: 0;
  border-radius: 50%;
}

<div
    class="c0"
  />
</DocumentFragment>
`;
